mantis-control-plane/mantis-control-plane-core/src/main/java/io/mantisrx/server/core/zookeeper/CuratorService.java

```diff
@@ -90,10 +83,11 @@ public void stateChanged(CuratorFramework client, ConnectionState newState) {
     @Override
     public void start() {
         try {
-            isConnectedGauge.set(0L);
-            setupCuratorListener();
-            curator.start();
-            masterMonitor.startAsync().awaitRunning();
```

Comment:
@Andyz26 - re: thread safety: wdyt of wrapping this in `synchronized (this) { ... }` to synchronize concurrent calls to `start()`?